encoder.go

```diff
@@ -0,0 +1,414 @@
+package rdb
+
+import (
+	"fmt"
+	"math"
+	"sync/atomic"
+	"time"
+)
+
+type Encoder struct {
```

Comment:
minor: maybe we can rename this to FileEncoder? I felt a bit puzzled when using this in collection encoders, when both the top level object is called encoder, as well as this